trl/extras/best_of_n_sampler.py

```diff
@@ -0,0 +1,74 @@
+from typing import Any, Callable, List, Optional, Union
+
+import torch
+from transformers import GenerationConfig, PreTrainedTokenizer, PreTrainedTokenizerFast
+
+from ..core import set_seed
+from ..models import SUPPORTED_ARCHITECTURES, PreTrainedModelWrapper
+
+
+class BestOfNSampler(object):
+    def __init__(
+        self,
+        model: PreTrainedModelWrapper,
+        tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
+        queries_to_scores: Callable[
+            [List[str]], List[float]
+        ],  # callable that takes a list of generated texts and returns the associated reward scores
```

Comment:
with docstrings we can remove comment

```suggestion
        ],
```